panic.go

```diff
@@ -0,0 +1,42 @@
+package slog
+
+import "fmt"
+
+// PanicError is a caught error from a panic() call
+type PanicError interface {
+	Error() string
+	Recovered() interface{}
+}
+
+// Recover attempts to recover from a panic
+func Recover() PanicError {
+	if rvr := recover(); rvr == nil {
+		// no panic
+		return nil
+	} else if p, ok := rvr.(PanicError); ok {
+		// pass through
+		return p
+	} else {
+		// wrap it
+		return &panicError{rvr}
+	}
+}
+
+type panicError struct {
+	payload interface{}
```

Comment:
`recover()` returns `any`, is there any reason to keep using `interface{}`?

Reply:
amended